### PR TITLE
[SYCL] Adjust for all Dims offset in accessor's device __init

### PR DESCRIPTION
**Pull request by aelovikov-intel**
The optimization done for 1-dim accessor is suitable for all dimensions.

The test sycl/test/gdb/accessors-device.cpp had to be updated as its previous 
implementation was fragile in capturing info it tried to verify.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
@elizabethandrews , can you please look at the test's change?

**Comment by aelovikov-intel** [comment]
Failures on 
```
Failed Tests (1):
  SYCL :: Basic/code_location_e2e.cpp
```

are unrelated and have been addressed already.

**Comment by aelovikov-intel** [comment]
@sergey-semenov , @elizabethandrews , gentle ping.

File: sycl/include/sycl/accessor.hpp [hunk 1]
```diff
@@ -832,17 +848,12 @@ class __SYCL_SPECIAL_CLASS accessor :
...
 
-#ifdef __SYCL_DEVICE_ONLY__
-    // Pointer is already adjusted for 1D case.
-    if (Dimensions == 1)
-      return Id[0];
-#endif // __SYCL_DEVICE_ONLY__
-
     size_t Result = 0;
-    // Unroll the following loop for both host and device code
-    __SYCL_UNROLL(3)
```
Comment (sergey-semenov) [inline sycl/include/sycl/accessor.hpp:843]: We probably don't need to keep `__SYCL_UNROLL` around then.

**Review by sergey-semenov — DISMISSED** [review]
The header part looks good to me. @elizabethandrews Could you please take a look at the test?

File: sycl/include/sycl/accessor.hpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/include/sycl/accessor.hpp:843]: Do you mean the definition of it?

Comment (sergey-semenov) [reply sycl/include/sycl/accessor.hpp:843]: Yes, that's what I meant.

File: sycl/test/gdb/accessors-device.cpp [hunk 2]
```diff
@@ -1,19 +1,51 @@
-// RUN: %clangxx -fsycl-device-only -c -fno-color-diagnostics -Xclang -ast-dump %s -I %sycl_include -Wno-sycl-strict | FileCheck %s
+// RUN: %clangxx -fsycl-device-only -c -fno-color-diagnostics -Xclang -fdump-record-layouts %s -I %sycl_include -Wno-sycl-strict | FileCheck %s
 // UNSUPPORTED: windows
-#include <sycl/accessor.hpp>
+#include <sycl/sycl.hpp>
 
 typedef sycl::accessor<int, 1, sycl::access::mode::read> dummy;
```
Comment (elizabethandrews) [inline sycl/test/gdb/accessors-device.cpp:5]: Is this being used anywhere?

Comment (aelovikov-intel) [reply sycl/test/gdb/accessors-device.cpp:5]: I think you're right, let me try to remove it.

**Review by elizabethandrews — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

**Comment by aelovikov-intel** [comment]
OCL failures 

```
********************
Failed Tests (1):
  SYCL :: XPTI/kernel/content.cpp

********************
Unexpectedly Passed Tests (2):

  SYCL :: SubGroup/reduce_spirv13.cpp
  SYCL :: SubGroup/reduce_spirv13_fp64.cpp
```

are known.

**Comment by aelovikov-intel** [comment]
@intel/dpcpp-esimd-reviewers , ping.

**Review by v-klochkov — APPROVED** [review]
Ok for ESIMD only part. 
I still have some questions for general/sycl part of the changes, but relying on @sergey-semenov opinion for @intel/llvm-reviewers-runtime part of the changes.

**Comment by aelovikov-intel** [comment]
> Ok for ESIMD only part. I still have some questions for general/sycl part of the changes, but relying on @sergey-semenov opinion for @intel/llvm-reviewers-runtime part of the changes.

Resolved via a call.

**Comment by aelovikov-intel** [comment]
@intel/llvm-gatekeepers , PR is ready.

**Pull request merged**